addons/mail/models/mail_activity.py

```diff
@@ -861,3 +861,70 @@ def activity_unlink(self, act_type_xmlids, user_id=None):
             domain = ['&'] + domain + [('user_id', '=', user_id)]
         self.env['mail.activity'].search(domain).unlink()
         return True
+
+    def _read_progress_bar(self, domain, group_by, progress_bar):
+        group_by_fname = group_by.partition(':')[0]
+        if not (progress_bar['field'] == 'activity_state' and self._fields[group_by_fname].store):
+            return super()._read_progress_bar(domain, group_by, progress_bar)
+
+        # optimization for 'activity_state'
```

Comment:
Should probably be a real docstring

Reply:
A docstring describes *what* the method does (specification), not *how* it does it (implementation).
The method's original docstring still applies.
This comment is about the implementation, not the specification.

Reply:
``# optimization for 'activity_state'`` is not a "how" (and actually it does not really helps anyone).